### PR TITLE
Add GCP KMS documentation

### PR DESCRIPTION
**Pull request by JoaoSa-da**

### CODE REVIEW

File: docs/2.7.0/docs/canton/architecture/security.rst [hunk 1]
```diff
@@ -141,9 +147,16 @@ The rotation frequency is fixed and cannot be changed.
...
-To do this Canton node administrators can request a :ref:`manual rotation of the KMS wrapper key through the Canton console <manual-aws-ksm-key-rotation>`.
+The manual rotation of a wrapper key requires not only the creation of a new KMS key but also the re-encryption of our data with it.
+To do this Canton node administrators can request a :ref:`manual rotation of the KMS wrapper key through the Canton console <manual-kms-wrapper-key-rotation>`.
+
+KMS Key Rotation
+^^^^^^^^^^^^^^^^
+
+When Canton's signing and encryption keys are off-sourced to a KMS (rather than encrypted at rest with a KMS wrapper key)
+their rotation has to be operated manually. Neither AWS or GCP provide automatic asymmetric key rotation.
+Nevertheless, this can be easily achieved by requesting a :ref:`standard rotation of Canton's keys <rotating-canton-keys>`,
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/architecture/security.rst:158]:
```suggestion
Manual key rotation is achieved by requesting a :ref:`standard rotation of Canton's keys <rotating-canton-keys>`,
```

File: docs/2.7.0/docs/canton/architecture/security.rst [hunk 2]
```diff
@@ -141,9 +147,16 @@ The rotation frequency is fixed and cannot be changed.
...
+which in this particular case will also involve the rotation of the underlying KMS key.
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/architecture/security.rst:159]:
```suggestion
which in this particular case also involves the rotation of the underlying KMS key.
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 3]
```diff
@@ -371,74 +387,108 @@ and then use its API to securely sign an decrypt messages. A Canton node still s
...
++-------------------------------------------+-------------------------------------------+
+| `kms:GetPublicKey`                        | `cloudkms.cryptoKeyVersions.viewPublicKey`|
++-------------------------------------------+-------------------------------------------+
 
 If you plan to use cross-account key usage then the permission for key rotation in Canton, namely `kms:CreateKey`, does not have to be configured as it does not apply in that use case.
 
-Canton uses the `standard AWS credential access
-<https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html>`_ to
-be able to make the API calls to the AWS KMS. For example, the standard
+To be able to make the API calls to the AWS KMS, Canton uses the `standard AWS credential access
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:479]:
```suggestion
To make the API calls to the AWS KMS, Canton uses the `standard AWS credential access
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 4]
```diff
@@ -371,74 +387,108 @@ and then use its API to securely sign an decrypt messages. A Canton node still s
...
+<https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html>`_. For example, the standard
 environment variables of `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` can
 be used. Alternatively, you can specify an AWS profile file (e.g. use a temporary access
-profile credentials - `sts`). The protection and rotation of
-the credentials for accessing AWS is a responsibility of the node operator.
+profile credentials - `sts`).
+
+For GCP, Canton uses a `GCP service account
+<https://cloud.google.com/docs/authentication/provide-credentials-adc#local-user-cred>`_. For example,
+standard environment variable `GOOGLE_APPLICATION_CREDENTIALS` can be used after
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:487]:
```suggestion
the standard environment variable `GOOGLE_APPLICATION_CREDENTIALS` can be used after
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 5]
```diff
@@ -371,74 +387,108 @@ and then use its API to securely sign an decrypt messages. A Canton node still s
...
+setting up a local Application Default Credentials (ADC) file for our service account.
+
+The protection and rotation of
+the credentials for accessing AWS or GCP are a responsibility of the node operator.
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:491]:
```suggestion
the credentials for AWS or GCP are the responsibility of the node operator.
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 6]
```diff
@@ -564,22 +627,41 @@ For example for a participant we would run:
...
+be aware that using the key id is not enough and we are required to register the key using its `ARN`.
 
 Finally, we need to initialize our :ref:`domain <manually-init-domain>` and
 :ref:`participants <manually-init-participant>` using the previously registered keys.
 
-.. _manual-aws-ksm-key-rotation:
+Manual KMS key rotation
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Canton keys can still be manually rotated even if externally stored in a KMS.
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:642]:
```suggestion
Canton keys can still be manually rotated even if they are externally stored in a KMS.
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 7]
```diff
@@ -564,22 +627,41 @@ For example for a participant we would run:
...
+To do that we can use the same :ref:`standard rotate
+key commands <rotating-canton-keys>` or, if we already have a KMS key to rotate to, run the following command:
+
+.. literalinclude:: /canton/includes/mirrored/enterprise/app/src/test/scala/com/digitalasset/canton/integration/tests/security/RotateAwsKmsKeyIntegrationTest.scala
+   :language: scala
+   :start-after: user-manual-entry-begin: RotateKmsNodeKey
+   :end-before: user-manual-entry-end: RotateKmsNodeKey
+   :dedent:
+
+Both AWS and GCP do not offer automatic rotation of asymmetric keys so, contrary to the wrapper key rotation,
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:652]:
```suggestion
Neither AWS or GCP offer automatic rotation of asymmetric keys so, unlike the wrapper key rotation,
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 8]
```diff
@@ -564,22 +627,41 @@ For example for a participant we would run:
...
+the node operator needs to be responsible for periodically rotating the keys.
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:653]:
```suggestion
the node operator needs to be responsible for periodically rotating these keys.
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 9]
```diff
@@ -564,22 +627,41 @@ For example for a participant we would run:
...
-AWS provides  tools to monitor KMS keys. To set automatic external logging, refer to the `official documentation
-<https://docs.aws.amazon.com/kms/latest/developerguide/monitoring-overview.html>`_.
+AWS and GCP provide tools to monitor KMS keys. For AWS to set automatic external logging, refer to the
+`AWS official documentation <https://docs.aws.amazon.com/kms/latest/developerguide/monitoring-overview.html>`_.
 This includes instructions on how to set AWS Cloud Trail or Cloud Watch Alarms
-to keep track of usage of KMS keys or of performed crypto operations. Errors resulting from the use of
-KMS keys are logged in Canton.
+to keep track of usage of KMS keys or of performed crypto operations.
+For GCP you can refer to the `GCP official documentation
+<https://cloud.google.com/kms/docs/audit-logging>`_ for information on how logging is done.
```
Comment (carrielaben-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:663]:
```suggestion
<https://cloud.google.com/kms/docs/audit-logging>`_ for information on logging.
```

**Review by carrielaben-da — APPROVED** [review]
Just a few small changes.

File: docs/2.7.0/docs/canton/architecture/security.rst [hunk 10]
```diff
@@ -40,6 +40,9 @@ Canton can, :ref:`if enabled <kms_setup>`, offer confidentiality at rest for the
 Short-term keys do not require additional protection because they are derived from a secret that is already
 transmitted and stored in an encrypted form using a long-term public encryption key.
 
+Long-term keys should include an operational security policy that requires to rotate these keys periodically
```
Comment (soren-da) [inline docs/2.7.0/docs/canton/architecture/security.rst:43]:
```suggestion
Long-term keys should be governed by an operational security policy that requires to rotate these keys periodically
```

File: docs/2.7.0/docs/canton/architecture/security.rst (hunk 1 above)
Comment (soren-da) [inline docs/2.7.0/docs/canton/architecture/security.rst:158]: you should add the caveat that this only works when canton generates the new keys automatically, not when the new key has already been generated in the KMS (the GS custodian model). we need to document the latter too, can be in a separate PR if not already done.

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 11]
```diff
@@ -74,7 +74,7 @@ if you wish to know how Canton can protect private keys whilst they remain inter
 :ref:`Externalize Private Keys With a Key Management Service <kms_external_architecture>`
 for more details on how Canton can enable private keys to be generated and stored by an external KMS.
 
-The following section :ref:`Key Management Service Setup <kms_setup>` describes how to enable AWS KMS for Canton
+The following section :ref:`Key Management Service Setup <kms_setup>` describes how to enable KMS for Canton
```
Comment (soren-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:77]:
```suggestion
The following section :ref:`Key Management Service Setup <kms_setup>` describes how to enable KMS support in Canton
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 12]
```diff
@@ -470,10 +524,20 @@ Please be aware that an AWS KMS key needs to be configured with the following se
 - Key specification: `SYMMETRIC_DEFAULT <https://docs.aws.amazon.com/kms/latest/developerguide/asymmetric-key-specs.html>`_
 - Key usage: `ENCRYPT_DECRYPT <https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#symmetric-cmks>`_
 
-If no ``wrapper-key-id`` is specified Canton creates a symmetric key in the KMS. After subsequent restarts the operator does not need to specify the identifier for the newly
+Similarly, for GCP KMS we can use:
+
+- Key name: `“test-key”`
```
Comment (soren-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:529]:
```suggestion
- Key name: `test-key`
```

File: docs/2.7.0/docs/canton/usermanual/security.rst [hunk 13]
```diff
@@ -470,10 +524,20 @@ Please be aware that an AWS KMS key needs to be configured with the following se
...
+- Key RN (Resource Name): `“projects/gcp-kms-testing/locations/us-east1/keyRings/canton-test-keys/cryptoKeys/test-key/cryptoKeyVersions/1”`
```
Comment (soren-da) [inline docs/2.7.0/docs/canton/usermanual/security.rst:530]:
```suggestion
- Key RN (Resource Name): `projects/gcp-kms-testing/locations/us-east1/keyRings/canton-test-keys/cryptoKeys/test-key/cryptoKeyVersions/1`
```

**Review by soren-da — APPROVED** [review]
